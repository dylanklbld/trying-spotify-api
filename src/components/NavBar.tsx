import {NavLink, useLocation, useRouteMatch} from 'react-router-dom'

import React from 'react'
import styled  from "styled-components";

const navbarColor = `#3B5765`

const Nav = styled.div`
  display: flex;
  flex-flow: column nowrap;
  align-items: center;
  font-size: 2em;
  text-align: center;
  font-family: Verdana,Geneva,sans-serif; 
  color: white;
`;

const NavLinksWrapper = styled.ul`
    margin: 0;
    padding: 0;
    text-align:center;
    list-style: none;
    list-style-type: none;
    height:100%;
    background: ${navbarColor};
`

const NavLinkStyled = styled(props=><NavLink {...props}/>)`
  text-decoration: none;
  font-size: 1em;
  transition: font-size 0.5s linear, text-shadow 1s;
  color: #09212D;
  &.active {
    font-size: 1.5em;
    color: white;
  }
  &:hover {
      cursor:pointer;
      color: white;
      text-shadow: 2px 2px 6px #0e98de;
  }
`;

const MenuItem = styled.li`
    display:inline-block;
    padding: 20px 20px;
`;

type Props = {
    links: any
}

export default function NavBar({links}: Props): React.ReactElement{
    const match = useRouteMatch()
    const { pathname } = useLocation()

    return (<Nav>
        <NavLinksWrapper>
          {links.map(({link, text}:any) => (
            <MenuItem key={link}>
                <NavLinkStyled to={`${match.path}/${link}`}
                    isActive={()=>[`${match.path}/${link}`].includes(pathname)}
                >
                    {text}
                </NavLinkStyled>
            </MenuItem>
            ))}
        </NavLinksWrapper>
    </Nav>)
}