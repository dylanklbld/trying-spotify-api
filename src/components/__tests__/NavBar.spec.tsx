import { fireEvent, render, screen } from '@testing-library/react';

import NavBar from '../NavBar';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useRouteMatch: () => ({ path: 'dashboard' }),
    useLocation:()=>({ pathname:'dashboard/path' })
}));


describe('NavBar', () => {
  it('should render empty NavBar', () => {
    const { container } = render(<Router>
        <NavBar links={[]}/>
        </Router>)
   
    expect(container).toMatchSnapshot()
  })
  
  it('should render NavBar with links', () => {
    const navigationLinks = [
        {
            link:`a-b-c`,
            text: "A B C"
        },{
            link:`test`,
            text: "TEST"
        }
    ]
    
    const { container } = render(<Router><NavBar links={navigationLinks}/></Router>)
   
    expect(container).toMatchSnapshot()
  })

  it('should have specific active link', () => {
    const navigationLinks = [
        {
            link:`a-b-c`,
            text: "A B C"
        },{
            link:`path`,
            text: "active-TEST"
        }
    ]
    
    const { container } = render(<Router><NavBar links={navigationLinks}/></Router>)
   
    expect(container).toMatchSnapshot()
  })
})