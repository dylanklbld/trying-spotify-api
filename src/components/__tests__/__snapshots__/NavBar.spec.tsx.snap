// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should have specific active link 1`] = `
<div>
  <div
    class="sc-bdfBwQ cCnEEX"
  >
    <ul
      class="sc-gsTCUz jdJIlU"
    >
      <li
        class="sc-hKgILt engOHZ"
      >
        <a
          class="sc-dlfnbm kUMyuz"
          href="/dashboard/a-b-c"
        >
          A B C
        </a>
      </li>
      <li
        class="sc-hKgILt engOHZ"
      >
        <a
          aria-current="page"
          class="sc-dlfnbm kUMyuz active"
          href="/dashboard/path"
        >
          active-TEST
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`NavBar should render NavBar with links 1`] = `
<div>
  <div
    class="sc-bdfBwQ cCnEEX"
  >
    <ul
      class="sc-gsTCUz jdJIlU"
    >
      <li
        class="sc-hKgILt engOHZ"
      >
        <a
          class="sc-dlfnbm kUMyuz"
          href="/dashboard/a-b-c"
        >
          A B C
        </a>
      </li>
      <li
        class="sc-hKgILt engOHZ"
      >
        <a
          class="sc-dlfnbm kUMyuz"
          href="/dashboard/test"
        >
          TEST
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`NavBar should render empty NavBar 1`] = `
<div>
  <div
    class="sc-bdfBwQ cCnEEX"
  >
    <ul
      class="sc-gsTCUz jdJIlU"
    />
  </div>
</div>
`;
